src/test/sighash_tests.cpp

```diff
@@ -90,30 +89,30 @@ uint256 static SignatureHashOld(CScript scriptCode, const CTransaction& txTo, un
 void static RandomScript(CScript &script) {
     static const opcodetype oplist[] = {OP_FALSE, OP_1, OP_2, OP_3, OP_CHECKSIG, OP_IF, OP_VERIF, OP_RETURN, OP_CODESEPARATOR};
     script = CScript();
-    int ops = (insecure_rand() % 10);
+    int ops = (InsecureRandRange(10));
     for (int i=0; i<ops; i++)
-        script << oplist[insecure_rand() % (sizeof(oplist)/sizeof(oplist[0]))];
+        script << oplist[InsecureRandRange(sizeof(oplist)/sizeof(oplist[0]))];
 }
 
 void static RandomTransaction(CMutableTransaction &tx, bool fSingle) {
-    tx.nVersion = (insecure_rand() % 2) + 1;
+    tx.nVersion = (InsecureRandBool()) + 1;
```

Comment:
This doesn't look correct. The intention here was not to get a random bool, but either a 1 or a 2, so the correct thing would now be `InsecureRandRange(2) + 1`

Reply:
https://github.com/dashpay/dash/pull/3021/commits/8c6f51ee19464d62e282246a8e3195aaefd06d39